docs/PUBLISHING.md

```diff
@@ -4,6 +4,48 @@ Publishing Etcher
 This is a small guide to package and publish Etcher to all supported operating
 systems.
 
+Release Types
+-------------
+
+Etcher supports **production** and **snapshot** release types. Each is
+published to a different S3 bucket, and production release types are code
+signed, while snapshot release types aren't and include a short git commit-hash
+as a build number. For example, `1.0.0-beta.19` is a production release type,
+while `1.0.0-beta.19+531ab82` is a snapshot release type.
+
+In terms of comparison: `1.0.0-beta.19` (production) < `1.0.0-beta.19+531ab82`
+(snapshot) < `1.0.0-rc.1` (production) < `1.0.0-rc.1+7fde24a` (snapshot) <
+`1.0.0` (production) < `1.0.0+2201e5f` (snapshot). Keep in mind that if you're
+running a production release type, you'll only be prompted to update to
+production release types, and if you're running a snapshot release type, you'll
+only be able to update to other snapshot release types.
+
+The build system creates (and publishes) snapshot release types by default, but
+you can build a specific release type by setting the `RELEASE_TYPE` make
+variable.  For example:
+
+```sh
+make <target> RELEASE_TYPE=snapshot
+make <target> RELEASE_TYPE=production
+```
+
+We can control the version range a specific Etcher version will consider when
+showing the update notification dialog by tweaking the `updates.semverRange`
+property of `package.json`.
+
+Update Channels
+---------------
+
+Etcher has a setting to include the unstable update channel. If this option is
+set, Etcher will consider both stable and unstable versions when showing the
+update notifier dialog. Unstable versions are the ones that contain a `beta`
+pre-release tag. For example:
+
+- Production stable version: `1.4.0`
+- Production unstable version: `1.4.0-beta.1`
+- Snapshot stable version: `1.4.0+7fde24a`
+- Snapshot unstable version: `1.4.0-beta.1+7fde24a`
```

Comment:
I wonder if ordering these according to their version-precedence would make things more or less confusing?

Production unstable version: `1.4.0-beta.1`
Snapshot unstable version: `1.4.0-beta.1+7fde24a`
Production stable version: `1.4.0`
Snapshot stable version: `1.4.0+7fde24a`